packages/core/elasticsearch/core-elasticsearch-server-internal/src/elasticsearch_service.ts

```diff
@@ -172,13 +174,15 @@ export class ElasticsearchService
     clientConfig: Partial<ElasticsearchClientConfig> = {}
   ) {
     const config = mergeConfig(baseConfig, clientConfig);
+
     return new ClusterClient({
       config,
       logger: this.coreContext.logger.get('elasticsearch'),
       type,
       authHeaders: this.authHeaders,
       getExecutionContext: () => this.executionContextClient?.getAsHeader(),
       getUnauthorizedErrorHandler: () => this.unauthorizedErrorHandler,
+      agentManager: this.agentManager,
```

Comment:
NIT: update https://github.com/elastic/kibana/blob/35083504464a4950d5ce5c6464cab387dd98d0d8/packages/core/elasticsearch/core-elasticsearch-server-internal/src/elasticsearch_service.test.ts#L124 accordingly to assert the agentManager is passed down to the `ClusterClient` constructor